### PR TITLE
refactor: descriptions cssinjs

### PR DESCRIPTION
**Pull request by Dunqing**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     --      |
| 🇨🇳 Chinese |     --      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/descriptions/Cell.tsx [hunk 1]
```diff
@@ -55,23 +58,30 @@ const Cell: React.FC<CellProps> = ({
 
   return (
     <Component
-      className={classNames(`${itemPrefixCls}-item`, className)}
+      className={classNames(`${itemPrefixCls}-item`, className, hashId)}
       style={style}
       colSpan={span}
     >
-      <div className={`${itemPrefixCls}-item-container`}>
+      <div className={classNames(`${itemPrefixCls}-item-container`, hashId)}>
```
Comment (zombieJ) [inline components/descriptions/Cell.tsx:65]: `hashId` 只需要在顶层 dom 添加，下面的子节点都不需要哈~

**Comment by 1247748612** [comment]
@zombieJ done

File: components/descriptions/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,242 @@
...
+        lineHeight: token.lineHeight,
+        textAlign: `start`,
+
+        '&::after': {
+          content: descriptionItemTrailingColon ? '":"' : '" "',
+          position: 'relative',
+          top: '-0.5px',
+          margin: `0 ${descriptionsItemLabelColonMarginRight}px 0 ${descriptionsItemLabelColonMarginLeft}px`,
+
+          [`${prefixCls}-rtl&`]: {
```
Comment (zombieJ) [inline components/descriptions/style/index.tsx:150]: rtl 的都通过 css logic props 代替：https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties

File: components/descriptions/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,242 @@
...
+        color: token.colorText,
+        fontWeight: 'normal',
+        fontSize: token.fontSize,
+        lineHeight: token.lineHeight,
+        textAlign: `start`,
+
+        '&::after': {
+          content: descriptionItemTrailingColon ? '":"' : '" "',
+          position: 'relative',
+          top: '-0.5px',
```
Comment (zombieJ) [inline components/descriptions/style/index.tsx:147]: 直接数字`-0.5`，另外所有直接写数字的地方都要加个 `// FIXME: hardcode in v4`~

File: components/descriptions/style/index.tsx (hunk 2 above)
Comment (zombieJ) [reply components/descriptions/style/index.tsx:150]: 其他地方用到过 left right 的相同

**Comment by zombieJ** [comment]
样式飞了：
截屏2022-03-31 下午3 19 28

页面右下角 dynamic theme 打开后勾选 diff 可以对比线上样式哈
截屏2022-03-31 下午3 20 19

**Comment by 1247748612** [comment]
> 样式飞了

done

**Pull request merged**